api/src/main/java/javax/enterprise/inject/spi/InjectionTargetFactory.java

```diff
@@ -56,4 +59,19 @@
      */
     public InjectionTarget<T> createInjectionTarget(Bean<T> bean);
 
+    /**
+     *
+     * Returns an {@link AnnotatedTypeConfigurator} to add or remove injection point to the {@link InjectionTarget} to create.
+     * {@link #createInjectionTarget(Bean)} will use the {@link AnnotatedTypeConfigurator} to define injection points in the returned
+     * {@link InjectionTarget}
+     *
+     * Each call returns the same AnnotatedTypeConfigurator.
+     *
+     * @return an {@link AnnotatedTypeConfigurator} to configure injection points
+     * @since 2.0
+     */
+    default AnnotatedTypeConfigurator<T> configure() {
+        return null;
```

Comment:
I agree that it's not nice, but look at the javadoc of the interface. It shows creation of an anonymous `InjectionTargetFactory` to wrap the one returned by the `BeanManager`. This code will break if user did that and update to 2.0...